src/serialize.h

```diff
@@ -513,6 +513,28 @@ struct VarIntFormatter
     }
 };
 
+template<int Bytes>
+struct CustomUintFormatter
+{
+    static_assert(Bytes > 0 && Bytes <= 8, "CustomUintFormatter Bytes out of range");
+    static constexpr uint64_t MAX = 0xffffffffffffffff >> (8 * (8 - Bytes));
+
+    template <typename Stream, typename I> void Ser(Stream& s, I v)
+    {
+        if (v < 0 || v > MAX) throw std::ios_base::failure("CustomUintFormatter value out of range");
+        uint64_t raw = htole64(v);
```

Comment:
Late to the party here -- but this does not seem equivalent to the previous encoding mechanism.  In particular, if the host machine is big endian, the previous code and this change will produce different, incompatible results.

Reply:
Can you give an example of how it would fail?

htole64 converts a number in host native representation (= the value the uint64_t input has in C) to 64-bit LE representation. On LE systems htole64 is a no-op. On BE systems it performs a byteswap on the representation of the number.

Reply:
Oh, nevermind. False alarm.  The serializations end up equivalent, they just take different paths to get there. :)